runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/partition/SerializedPartition.java

```diff
@@ -59,7 +59,7 @@
   @Nullable
   private final EncoderFactory.Encoder encoder;
   private final MemoryPoolAssigner memoryPoolAssigner;
-  private volatile List<MemoryChunk> dataList;
+  private List<MemoryChunk> dataList;
```

Comment:
Is it okay to non-volatilize this object? It seems the volatilizing of the reference is intended.
If it is needed, you can just put a comment about it and ignore the SonarCloud issue.

Reply:
I suppose the reference does not change once it is assigned, either on commit() or construction, so it would okay to non-volatilize it. @DifferentSC could you take a look too?

Reply:
I think removing `volatile` is okay, because it seems that the reference to `dataList` doesn't change.

Reply:
@hy00nc 
IMHO, if `SerializedPartition` is constructed with the constructor in [L75](https://github.com/apache/incubator-nemo/pull/243/files#diff-435f870c711c13b5dc39a9b7a9248c74L75) and committed, the reference of `dataList` can be changed. Please let me know if I misunderstood.

Reply:
if SerializedPartition is constructed in that way, dataList is assigned only once on commit(), and used only after commit(). So I assume no inconsistency will occur?

Reply:
But the thread that uses this partition after the `commit()` can see that the `dataList` is `null` because the reference is not volatile, right?

Reply:
I got your point, thanks for your kind explanation! I will put the volatile back and update the PR soon👍